envoy-control-core/src/main/resources/lua/ingress_rbac_logging.lua

```diff
@@ -19,15 +19,11 @@ function envoy_on_response(handle)
 
     local lua_metadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.lua") or {}
     local service_name = lua_metadata["request.info.service_name"] or ""
-    local path = lua_metadata["request.info.path"]
-    path = path and "'"..path.."'" or ""
+    local path = lua_metadata["request.info.path"] or ""
     local protocol = handle:connection():ssl() == nil and "http" or "https"
     local method = lua_metadata["request.info.method"] or ""
     local xff_header = lua_metadata["request.info.xff_header"] or ""
     local source_ip = string.match(xff_header, '[^,]+$') or ""
-    local statusCode = handle:headers():get(":status") or ""
-    handle:logInfo("\nUnauthorized request: method = "..method..", path = "..path..", clientIp = "..source_ip..
-      ", clientName = "..service_name..", protocol = "..protocol..", statusCode = "..statusCode)
+    local statusCode = handle:headers():get(":status") or "0"
+    handle:logInfo("\nINCOMING_PERMISSIONS { \"method\": \""..method.."\", \"path\": \""..path.."\", \"clientIp\": \""..source_ip.."\", \"clientName\": \""..service_name.."\", \"protocol\": \""..protocol.."\", \"statusCode\": "..statusCode.." }")
```

Comment:
And the question appears: could any of the fields contain `"` character or any other characters prohibited inside a json? If yes, it will build an invalid JSON.
Maybe it would be easier, if we just use a newline as a fields separator, instead of trying to create a json?

Reply:
- https://tools.ietf.org/html/rfc3986#page-48 - does not define `"` as a valid character anywhere in the URL (so we should be safe in protocol and path)
- method definitions are enum - https://tools.ietf.org/html/rfc7231#section-4.3 they do not contain "
- ips - ipv4 is numbers separated by dots, ipv6 - separated by colon
- clientName comes from URI san so the same as path and protocol
- status code are positive integers https://tools.ietf.org/html/rfc7231#page-48

Reply:
* There are more characters to escape: https://www.freeformatter.com/json-escape.html. I would say that most risky are backslash and double quote

> clientName comes from URI san so the same as path and protocol

clientName comes from a header value, so I think right now this is the only risky value :(

What about simply printing each field in separate line? We will reduce number of prohibited characters to \n only

Reply:
OK, we talked about it and decided that consequences of invalid json will be acceptable. So we would keep it as it is.

Reply:
Do we drop any query params in the path? Query parameters might contains `'`/`"`.

Reply:
@lukidzi query params are included in the `path` field. But `/` is not a problem, it doesn't need to be escaped in json: https://www.freeformatter.com/json-escape.html
I think we are safe regarding `path` field, according to https://tools.ietf.org/html/rfc3986#page-48